core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -26,74 +34,97 @@
 /**
  * Container which launches Docker Compose, for the purposes of launching a defined set of containers.
  */
-public class DockerComposeContainer<SELF extends DockerComposeContainer<SELF>> extends GenericContainer<SELF> implements LinkableContainer {
+public class DockerComposeContainer<SELF extends DockerComposeContainer<SELF>> extends FailureDetectingExternalResource {
```

Comment:
This class is now a JUnit rule but not a descendent of GenericContainer. It is just responsible for launching a container that has docker-compose (see DockerCompose class) and the ambassador containers.

I would like to rename, but have left name as-is for backwards compatibility.